packages/wagmi/tests/client.test.ts

```diff
@@ -0,0 +1,100 @@
+    enableAnalytics: true,
+    metadata,
+    termsConditionsUrl: 'https://walletconnect.com/terms',
+    privacyPolicyUrl: 'https://walletconnect.com/privacy',
+    tokens: {
+      1: {
+        address: '0x123',
+        image: 'https://example.com/token.png'
+      }
+    },
+    customWallets: [
+      {
+        id: 'react-wallet-v2',
+        name: 'react-wallet-v2',
+        homepage: 'https://react-wallet.walletconnect.com/',
+        mobile_link: 'https://react-wallet.walletconnect.com/',
+        desktop_link: 'https://react-wallet.walletconnect.com/',
+        webapp_link: 'https://react-wallet.walletconnect.com/'
+      }
+    ],
+    allWallets: 'SHOW',
+    includeWalletIds: ['8a0ee50d1f22f6651afcae7eb4253e52a3310b90af5daef78a8c4929a9bb99d4'],
+    excludeWalletIds: ['c03dfee351b6fcc421b4494ea33b9d4b92a984f87aa76d1663bb28705e95034a'],
+    featuredWalletIds: ['1ae92b26df02f0abca6304df07debccd18262fdf5fe82daa81593582dac9a369'],
+    allowUnsupportedChain: true
+  })
+})
+
+// -- Tests --------------------------------------------------------------------
+describe('createWeb3Modal by wagmi', () => {
+  it('should set controller states as expected', () => {
+    expect(OptionsController.state.projectId).toBe('PROJECT_ID')
+    expect(OptionsController.state.allWallets).toBe('SHOW')
+    expect(OptionsController.state.includeWalletIds).toEqual([
+      '8a0ee50d1f22f6651afcae7eb4253e52a3310b90af5daef78a8c4929a9bb99d4'
+    ])
+    expect(OptionsController.state.excludeWalletIds).toEqual([
+      'c03dfee351b6fcc421b4494ea33b9d4b92a984f87aa76d1663bb28705e95034a'
+    ])
+    expect(OptionsController.state.featuredWalletIds).toEqual([
+      '1ae92b26df02f0abca6304df07debccd18262fdf5fe82daa81593582dac9a369'
+    ])
+    expect(OptionsController.state.tokens?.['eip155:1']?.address).toBe('0x123')
+    expect(OptionsController.state.termsConditionsUrl).toBe('https://walletconnect.com/terms')
+    expect(OptionsController.state.privacyPolicyUrl).toBe('https://walletconnect.com/privacy')
+    expect(OptionsController.state.enableAnalytics).toBe(true)
+    expect(OptionsController.state.customWallets?.[0]?.id).toEqual('react-wallet-v2')
+    expect(OptionsController.state.isUniversalProvider).toBe(undefined)
+    expect(OptionsController.state.sdkVersion).toEqual(`html-wagmi-${ConstantsUtil.VERSION}`)
+    expect(OptionsController.state.enableOnramp).toBe(true)
+    expect(OptionsController.state.metadata).toEqual(metadata)
+    expect(OptionsController.state.disableAppend).toEqual(undefined)
+    expect(NetworkController.state.allowUnsupportedChain).toBe(true)
+  })
+
+  it('should set connectors as expected', () => {
+    const authConnector = ConnectorController.state.connectors.find(c => c.type === 'AUTH')
+    expect(ConnectorController.state.connectors.length).toBeGreaterThan(0)
+    expect(authConnector).toBeDefined()
+    expect(authConnector?.socials).toEqual(['google', 'x', 'discord', 'apple', 'github'])
+  })
```

Comment:
should we do some combinations of connectors to check if they are properly set? 
Also, would check for CB connector

Reply:
We should def do that, I've tried this in the TDW but was more complex then I thought. What I want to do is that refactoring whole these tests when we switch to new architecture. We'll need to remove these files and move to adapters, and we could refactor the tests in that time as well.